api/data_explorer/__main__.py

```diff
@@ -268,24 +259,31 @@ def _get_es_facets():
             # is fixed, use AutoHistogramFacet instead. Will no longer need 2
             # steps.
             max_field_value = _get_max_field_value(es, field_name)
-            facets[ui_facet_name] = HistogramFacet(
+            es_facets[ui_facet_name] = HistogramFacet(
                 field=field_name, interval=_get_interval(max_field_value))
-    app.app.logger.info('Elasticsearch facets: %s' % facets)
-    return facets
+
+    app.app.logger.info('Elasticsearch facets: %s' % es_facets)
+    app.app.config['ELASTICSEARCH_FACETS'] = es_facets
+    app.app.logger.info('UI facets: %s' % ui_facets)
+    app.app.config['UI_FACETS'] = ui_facets
```

Comment:
opt - Elasticsearch field name isn't related to UI. Would be a bit cleaner to have just one facet dict with everything. This is optional, and can be in a different PR